feat(ui): add ShareTimeRangeButton component

eskirk opened the pull request:
moving from https://github.com/grafana/grafana/pull/100940 to here 

**What is this feature?**

this feature adds a reusable "Share Time Range" button component to the grafana-ui package. this button shares the current URL with the `from` and `to` query parameters changed from relative time to absolute time. 

**Why do we need this feature?**

many plugins and other areas of Grafana leverage relative time for basic use, but when it comes to sharing that view with others, copying the URL falls short since the link effectively expires after a a short time. 

this button provides a universal way of sharing a snapshot of time easily and across many areas of Grafana. 

**Who is this feature for?**

developers who want to make it easier to URLs in their plugins 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

<img width="700" alt="image" src="https://github.com/user-attachments/assets/e7b8a63b-428d-43bd-a9e0-14d854f2b1f9" />

eskirk committed:
feat(ui): add ShareTimeRangeButton component

ivanahuckova commented on src/components/ShareTimeRangeButton/ShareTimeRangeButton.tsx:
```diff
@@ -0,0 +1,110 @@
+
+import { type GrafanaTheme2 } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+import { copyText } from '../../utils/clipboard';
+import { t, Trans } from '../../utils/i18n';
+import { absoluteTimeRangeURL } from '../../utils/time';
+import { Button, ButtonGroup, type ButtonProps } from '../Button';
+import { ClipboardButton } from '../ClipboardButton/ClipboardButton';
+import { Dropdown } from '../Dropdown/Dropdown';
+import { Menu } from '../Menu/Menu';
+import { type MenuItemElement } from '../Menu/MenuItem';
```
These all are no longer accessible through relative path and you need to import them from `@grafana/ui`

ivanahuckova commented on src/components/ShareTimeRangeButton/ShareTimeRangeButton.tsx:
```diff
@@ -0,0 +1,110 @@
+import { css } from '@emotion/css';
+import { useRef, useState } from 'react';
```
```suggestion
import React, { useRef, useState } from 'react';
```

ivanahuckova reviewed (changes requested):
Left some comments bellow - mainly related to relative imports that should be now imported in grafana/ui, as these are not going to be accessible in this package anymore. 

I am also wondering, if you've tried to build and test this locally because currently it throws a lot of errors

<img width="1483" alt="image" src="https://github.com/user-attachments/assets/651a3545-2ed8-4cb2-b8cf-d1126a191390" />

eskirk commented:
hey @ivanahuckova sorry I have converted this to a draft. I have not tried building/running it in this repo, I just moved this over from a `grafana/grafana` PR and have not had time to work on it. should be able to do that this week. 

thanks for taking a look 🤝

ivanahuckova commented:
I am closing as there hasn't been any activity in a while
